model_clone/mixins/clone.py

```diff
@@ -321,12 +321,16 @@ def _create_copy_of_instance(instance, using=None, force=False, sub_clone=False)
         new_instance = cls()
 
         for f in fields:
-            value = f.get_default() or getattr(instance, f.attname, None)
-
+            # Use default value for special fields like timestamp, uuid
             if isinstance(f, (models.DateTimeField, models.DateField)):
                 if f.auto_now or f.auto_now_add:
                     f.pre_save(new_instance, add=True)
                     continue
+            if not f.editable and f.name in unique_fields:
+                f.pre_save(new_instance, add=True)
+                continue
+
+            value = getattr(instance, f.attname, f.get_default())
```

Comment:
Can we fallback to the default.

```
value = getattr(instance, f.attname, None) or  f.get_default()
```

Reply:
So for `""`, `0` and `None` values, we would use default value? I'm not sure if this is a "fallback"?

Reply:
Problem here is the default wouldn’t be called since the source instance would always have the attribute the fallback in the case would be to use the default value if the current source is None

Reply:
Sorry, please correct me if I'm wrong: the `fields` we got from `get_fields_and_unique_fields_from_cls` is all the fields that potentially need to be cloned, which means no matter what value it is(as long as it's valid), it should be cloned to the new instance.
I just added a test case to demonstrate my idea, and removed the fallback default value -- as you pointed out, it would never really fallback.

Reply:
Yeah I agree based on this change using the default wouldn’t make much sense if we always want to use the source instance values.